src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs

```diff
@@ -320,64 +329,85 @@ public static void Update()
                         state.rawValue *= -1.0f;
                     }
                     state.smoothedValue = state.rawValue; // smoothing is not applied on mouse movement
-                } 
+                }
                 else if (entry.type == InputManagerEntryType.JOYSTICK_AXIS)
                 {
                     // joysticks unsupported currently
                     // eventually this will read the joystick value, incorporating entry.dead for dead zone
-                } 
+                }
                 else
                 {
                     RGDebug.LogWarning($"Unexpected input manager entry type: {entry.type}");
                 }
             }
-                
+
             // Invoke the appropriate mouse event handlers on MonoBehaviours
             {
-                (GameObject, Camera)? hitObject = null;
-                (GameObject, Camera)? hitObject2D = null;
                 int numCameras = Camera.allCamerasCount;
                 if (_camerasBuf == null || _camerasBuf.Length != numCameras)
                 {
                     _camerasBuf = new Camera[numCameras];
                 }
+
                 Camera.GetAllCameras(_camerasBuf);
                 foreach (Camera camera in _camerasBuf)
                 {
                     if (camera == null || camera.eventMask == 0 || camera.targetTexture != null)
                     {
                         continue;
                     }
-                        
-                    int cameraRaycastMask = camera.cullingMask & camera.eventMask;
-                        
+
+                    var cameraRaycastMask = camera.cullingMask & camera.eventMask;
+
+                    (GameObject, GameObject) result = (null, null);
                     // 3D raycast
                     {
                         Ray mouseRay = camera.ScreenPointToRay(_mousePosition);
                         if (Physics.Raycast(mouseRay, out RaycastHit hit, maxDistance: Mathf.Infinity,
                                 layerMask: cameraRaycastMask))
                         {
-                            hitObject = (hit.collider.gameObject, camera);
+                            _currentHitObject = (hit.collider.gameObject, camera);
                         }
                     }
-                        
+
                     // 2D raycast
                     {
                         Vector3 mouseWorldPt = camera.ScreenToWorldPoint(new Vector3(_mousePosition.x, _mousePosition.y, camera.nearClipPlane));
-                        RaycastHit2D hit2D = Physics2D.Raycast(mouseWorldPt, Vector2.zero, distance: Mathf.Infinity, 
-                            layerMask: cameraRaycastMask);
-                        if (hit2D.collider != null)
+                        var collider = Physics2D.OverlapPoint(mouseWorldPt, cameraRaycastMask);
```

Comment:
I'll make sure to do some extensive testing in my UI testing UAT to make sure this works as expected 👍

Reply:
Reminder, this only affects Legacy InputManager games